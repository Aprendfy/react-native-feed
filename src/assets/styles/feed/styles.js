import { StyleSheet, Platform } from 'react-native';
import { colors, hm } from '../styles';

export const cardHeaderStyles = StyleSheet.create({
  mainContainer: {
    flex: 1
  },
  imageCover: {
    flex: 1,
    height: 25 * hm
  },
  firstContainer: {
    flex: 1,
    paddingHorizontal: 10
  },
  titleWrapper: {
    flex: 1,
    justifyContent: 'flex-end'
  },
  title: {
    color: colors.whitePrimary,
    fontSize: 27,
    fontWeight: '500'
  },
  secondContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10
  },
  itemDetailWrapper: {
    flex: 1,
    flexDirection: 'row'
  },
  detailText: {
    marginHorizontal: 5,
    color: colors.whitePrimary,
    fontSize: 12
  }
});
