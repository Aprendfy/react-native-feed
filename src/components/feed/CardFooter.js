import React from 'react';
import { AdMobBanner } from 'react-native-admob';

export const CardFooter = () => {
  return (
    <AdMobBanner
      bannerSize="smartBannerPortrait"
      adUnitID="ca-app-pub-5511910440808215/9505833480"
      testDeviceID="EMULATOR"
      didFailToReceiveAdWithError={error => console.log(error, 'error de banner')}
    />
  );
};
