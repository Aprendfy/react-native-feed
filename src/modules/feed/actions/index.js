import { SAVE_FEED } from './types';
import { fakeData } from '../../../config/mockData';

export function fetchFeed(category) {
  return async (dispatch) => {
    const items = await fakeData(category);
    dispatch(saveFeed(items));
  };
}

export function saveFeed(payload) {
  return {
    type: SAVE_FEED,
    payload
  };
}
