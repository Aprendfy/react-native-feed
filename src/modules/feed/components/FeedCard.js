import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import { CardHeader, CardBody, CardFooter } from './';
import Styles from './styles/FeedCardStyle';

export class FeedCard extends Component {
  constructor(props) {
    super(props);
    this.state = { isCollapsed: true };
  }

  toggleIsCollapsed() {
    this.state.isCollapsed = !this.state.isCollapsed;
  }

  render() {
    const { isCollapsed } = this.state;
    const { title, category, readingTime, level, color, image, body } = this.props;
    const headerProps = { title, readingTime, level, color, image, category };

    return (
      <View style={isCollapsed ? Styles.collapsedContainer : Styles.fullContainer}>
        <CardHeader {...headerProps} />
        <CardBody text={body} />
        <CardFooter />
      </View>
    );
  }
}

FeedCard.propTypes = {
  title: PropTypes.string,
  category: PropTypes.string,
  readingTime: PropTypes.string,
  level: PropTypes.string,
  color: PropTypes.string,
  image: PropTypes.string,
  body: PropTypes.string,
};

FeedCard.defaultProps = {
  title: '',
  category: '',
  readingTime: '',
  level: '',
  color: '',
  image: '',
  body: '',
};
