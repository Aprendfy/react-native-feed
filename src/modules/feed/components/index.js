export { CardBody } from './CardBody';
export { CardAdsBanner } from './CardAdsBanner';
export { CardHeader } from './CardHeader';
export { CardSeparator } from './CardSeparator';
export { FeedCard } from './FeedCard';
export { FeedList } from './FeedList';
