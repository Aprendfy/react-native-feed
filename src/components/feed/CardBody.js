import React from 'react';
import PropTypes from 'prop-types';

import { Text, View } from 'react-native';
import { cardBodyStyle as styles } from '../../assets/styles/feed/styles';

export const CardBody = ({ text }) => {
  return (
    <View style={styles.textWrapper}>
      <Text style={styles.text}>
        {text}
      </Text>
    </View>
  );
};

CardBody.propTypes = {
  text: PropTypes.string.isRequired,
};
