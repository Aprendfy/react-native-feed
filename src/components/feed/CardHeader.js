import React, { Component } from 'react';
import { Text, View, Image } from 'react-native';
import IonIcons from 'react-native-vector-icons/Ionicons';
import LinearGradient from 'react-native-linear-gradient';
import { colors } from '../../assets/styles/styles';

class CardHeader extends Component {

  constructor(props) {
    super(props);
  }

  render() {
    const { title, tag, time, level, color, image } = this.props;
    return (
      <View style={{ flex: 1 }}>
        <Image
          style={{ flex: 1 }}
          resizeMethod="scale"
          resizeMode="cover"
          source={image}
        >
          <LinearGradient colors={['transparent', color]} style={{ flex: 1, paddingHorizontal: 10 }}>
            <View style={{ flex: 1, justifyContent: 'flex-end' }}>
              <Text style={{ color: 'white', fontSize: 27, fontWeight: '500' }}>{title}</Text>
            </View>
          </LinearGradient>
          <View style={{ flexDirection: 'row', justifyContent: 'space-around', backgroundColor: color, padding: 10 }}>
            <View style={{ flex: 1, flexDirection: 'row' }}>
              <IonIcons name="ios-pricetag-outline" size={14} color={colors.whitePrimary} />
              <Text style={{ marginHorizontal: 5, color: colors.whitePrimary, fontSize: 12 }}>{tag}</Text>
            </View>
            <View style={{ flex: 1, flexDirection: 'row' }}>
              <IonIcons name="ios-time-outline" size={14} color={colors.whitePrimary} />
              <Text style={{ marginHorizontal: 5, color: colors.whitePrimary, fontSize: 12 }}>{time}</Text>
            </View>
            <View style={{ flex: 1, flexDirection: 'row' }}>
              <IonIcons name="ios-school" size={14} color={colors.whitePrimary} />
              <Text style={{ marginHorizontal: 5, color: colors.whitePrimary, fontSize: 12 }}>{level}</Text>
            </View>
          </View>

        </Image>

      </View>

    );
  }

}

export default CardHeader;
