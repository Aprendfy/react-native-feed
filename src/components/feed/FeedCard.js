import React, { Component, PropTypes } from 'react';
import { Text, View } from 'react-native';

import { AdMobBanner } from 'react-native-admob';

import { colors } from '../../assets/styles/styles';
import CardHeader from './CardHeader';

class FeedCard extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { title, category, readingTime, level, color, image, body, dateCreated } = this.props;
    return (
      <View style={{ flex: 1 }}>
        <View style={{ flex: 2 }}>
          <CardHeader
            title={title}
            tag={category}
            time={readingTime}
            level={level}
            color={color}
            image={{ uri: image }}
          />
        </View>
        <View style={{ flex: 3 }}>
          <Text style={{ color: colors.blackPrimary, fontSize: 14, padding: 5 }}>
            {body}
          </Text>
        </View>
        <View style={{ position: 'absolute', left: 0, right: 0, bottom: 0 }}>
          <AdMobBanner
            bannerSize="smartBannerPortrait"
            adUnitID="ca-app-pub-8356555649836141/9541656259"
            testDeviceID="EMULATOR"
            didFailToReceiveAdWithError={this.bannerError}
          />
        </View>
      </View>
    );
  }
}

FeedCard.propTypes = {
  title: PropTypes.string,
  category: PropTypes.string,
  readingTime: PropTypes.string,
  level: PropTypes.string,
  color: PropTypes.string,
  image: PropTypes.string,
  body: PropTypes.string,
  dateCreated: PropTypes.string,

};

FeedCard.defaultProps = {
  title: '',
  category: '',
  readingTime: '',
  level: '',
  color: '',
  image: '',
  body: '',
  dateCreated: '',

};

export default FeedCard;
