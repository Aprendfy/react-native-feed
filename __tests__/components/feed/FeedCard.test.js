import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { FeedCard } from '../../../src/components/feed/FeedCard';

const props = {
  title: 'title',
  tag: 'tag',
  readingTime: 'readingtime',
  level: 'level',
  color: 'red',
  image: 'image',
  category: 'category',
  body: 'BODY'
};

describe('<FeedCard />', () => {
  it('Should have a snapshot', () => {
    const wrapper = shallow(<FeedCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
