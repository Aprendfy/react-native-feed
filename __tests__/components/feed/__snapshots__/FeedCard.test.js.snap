// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedCard /> Should have a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "flex": 1,
          }
    }
>
    <View
        style={
            Object {
                "flex": 2,
              }
        }
    >
        <CardHeader
            color="red"
            image={
                Object {
                    "uri": "image",
                  }
            }
            level="level"
            readingTime="readingtime"
            tag="category"
            title="title"
        />
    </View>
    <View
        style={
            Object {
                "flex": 3,
              }
        }
    >
        <CardBody
            text="BODY"
        />
    </View>
    <View
        style={
            Object {
                "backgroundColor": "#f5f5f5",
                "flex": 1,
                "justifyContent": "flex-end",
              }
        }
    >
        <CardFooter />
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "flex": 1,
                }
      }
>
      <View
            style={
                  Object {
                        "flex": 2,
                      }
            }
      >
            <CardHeader
                  color="red"
                  image={
                        Object {
                              "uri": "image",
                            }
                  }
                  level="level"
                  readingTime="readingtime"
                  tag="category"
                  title="title"
            />
      </View>
      <View
            style={
                  Object {
                        "flex": 3,
                      }
            }
      >
            <CardBody
                  text="BODY"
            />
      </View>
      <View
            style={
                  Object {
                        "backgroundColor": "#f5f5f5",
                        "flex": 1,
                        "justifyContent": "flex-end",
                      }
            }
      >
            <CardFooter />
      </View>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <FeedCard
        body="BODY"
        category="category"
        color="red"
        image="image"
        level="level"
        readingTime="readingtime"
        tag="tag"
        title="title"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": FeedCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "body": "BODY",
          "category": "category",
          "color": "red",
          "image": "image",
          "level": "level",
          "readingTime": "readingtime",
          "tag": "tag",
          "title": "title",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
>
          <View
                    style={
                              Object {
                                        "flex": 2,
                                      }
                    }
          >
                    <CardHeader
                              color="red"
                              image={
                                        Object {
                                                  "uri": "image",
                                                }
                              }
                              level="level"
                              readingTime="readingtime"
                              tag="category"
                              title="title"
                    />
          </View>
          <View
                    style={
                              Object {
                                        "flex": 3,
                                      }
                    }
          >
                    <CardBody
                              text="BODY"
                    />
          </View>
          <View
                    style={
                              Object {
                                        "backgroundColor": "#f5f5f5",
                                        "flex": 1,
                                        "justifyContent": "flex-end",
                                      }
                    }
          >
                    <CardFooter />
          </View>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
>
          <View
                    style={
                              Object {
                                        "flex": 2,
                                      }
                    }
          >
                    <CardHeader
                              color="red"
                              image={
                                        Object {
                                                  "uri": "image",
                                                }
                              }
                              level="level"
                              readingTime="readingtime"
                              tag="category"
                              title="title"
                    />
          </View>
          <View
                    style={
                              Object {
                                        "flex": 3,
                                      }
                    }
          >
                    <CardBody
                              text="BODY"
                    />
          </View>
          <View
                    style={
                              Object {
                                        "backgroundColor": "#f5f5f5",
                                        "flex": 1,
                                        "justifyContent": "flex-end",
                                      }
                    }
          >
                    <CardFooter />
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FeedCard
    body="BODY"
    category="category"
    color="red"
    image="image"
    level="level"
    readingTime="readingtime"
    tag="tag"
    title="title"
/>,
}
`;
